subnet/subnet_contract.go

```diff
@@ -61,15 +60,43 @@ func _deploy(data interface{}, isUpdate bool) {
 	if isUpdate {
 		args := data.([]interface{})
 		common.CheckVersion(args[len(args)-1].(int))
+
+		switchToNotary(storage.GetContext(), args)
+
 		return
 	}
 
 	args := data.(struct {
 		notaryDisabled bool
 	})
 
-	ctx := storage.GetContext()
-	storage.Put(ctx, []byte{notaryDisabledKey}, args.notaryDisabled)
+	if args.notaryDisabled {
+		panic("non-notary mode is not supported anymore")
+	}
+}
+
+// re-initializes contract from non-notary to notary mode. Does nothing if
+// action has already been done. The function is called on contract update with
+// storage.Context and parameters from _deploy.
+//
+// If contract stores non-empty value by 'ballots' key, switchToNotary panics.
+// Otherwise, existing value is removed.
+//
+// switchToNotary removes value stored by 'notary' key.
+func switchToNotary(ctx storage.Context, args []interface{}) {
+	const notaryDisabledKey = "notary" // non-notary legacy
+
+	notaryVal := storage.Get(ctx, notaryDisabledKey)
+	if notaryVal == nil {
+		runtime.Log("contract is already notarized")
+		return
+	}
+
```

Comment:
BTW, this contract can have voting data as well.

Reply:
How did you determine it? I don't see the corresponding code. For certainty?

Reply:
It uses `common.Vote()` in `Put()`